// Copyright (c) 2015 Chaobin Zhang. All rights reserved.
// Use of this source code is governed by the BSD license that can be
// found in the LICENSE file.

package rpc;

option cc_generic_services = true;

message RpcMessage {
  enum Type { 
    REQUEST  = 1;
    RESPONSE = 2;
  }

  required fixed64 id = 1;
  required Type type = 2;
  required string service = 3;
  required string method = 4;

  optional bytes request = 5;
  optional bytes response = 6;
}
